ui/components/app/currency-input/currency-input.js

```diff
@@ -101,17 +95,20 @@ export default function CurrencyInput({
         });
 
     setNewHexValue(hexValueNew);
-    setDecimalValue(newDecimalValue);
-    onChange(hexValueNew);
+    await onChange(hexValueNew);
```

Comment:
Can you help me understand why we are awaiting here? While there's nothing technically wrong with async event handlers, having  multiple that are passed between components seems like a great way to have unintended side effects such as memory leaks or attempts to update state in unmounted components. I did a fairly quick look into  this and it seems the only place this component is even used is in the UserPreferencedCurrencyInput component, which doesn't appear to ever be given an onChange callback. I could be wrong, so try and investigate and see if you can find where its used.

Reply:
@brad-decker Thank's for this catch! I investigate and searched in code where this component is used, and you are right, it is used only in `UserPreferencedCurrencyInput`. Because there is no attempts to update state in component  I removed `async/await` type and check if everything is working fine when switching between ETH and USD.
You can see changes [here](https://github.com/MetaMask/metamask-extension/pull/13827/commits/5890e75fad25cd7bb54f029bfcdc22b9c0baf3c7)